mono/mini/mini-amd64.c

```diff
@@ -392,6 +392,20 @@ add_valuetype_win64 (MonoMethodSignature *sig, ArgInfo *ainfo, MonoType *type,
 
 	klass = mono_class_from_mono_type (type);
 	size = mini_type_stack_size_full (&klass->byval_arg, NULL, sig->pinvoke);
+
+	ainfo->pass_empty_struct = FALSE;
+#ifndef __GNUC__
+	/*
```

Comment:
I'm not sure this ifdef is needed, the compiler used to compile the runtime might not be the same as the compiler used to compile the pinvoked code, so its better to just use the new behavior everywhere.